server/node/src/authz.js

```diff
@@ -0,0 +1,19 @@
+import { OPA } from "opa/highlevel/index.js";
+
+export class UnauthorizedError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "UnauthorizedError";
+  }
+}
+
+export class Authorizer {
+  constructor(endpoint) {
+    this.opa = new OPA(endpoint);
+  }
+
+  async authorized(path, input, { auth: { tenant, subject: user } }) {
```

Comment:
I go back and forth on weather I think the authz should just be a boolean allow/deny or if it should be some kind of exception. On the Java side, the API server does ultimately throw an unauthorized exception that gets consumed by the server framework.